homeassistant/components/onkyo/config_flow.py

```diff
@@ -0,0 +1,271 @@
+
+    async def async_step_manual(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Discover a receiver using a manually set host."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Run discovery for specified host.
+            self._connection = next(
+                iter(
+                    self.get_unique_connections(
+                        await async_discover_connections(host=user_input[CONF_HOST])
+                    )
+                ),
+                None,
+            )
+
+            if self._connection:
+                return await self.async_step_connect()
+
+            errors["base"] = "discovery_error"
+
+        return self.async_show_form(
+            step_id="manual", data_schema=MANUAL_SCHEMA, errors=errors
+        )
+
+    async def async_step_select(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle multiple network receivers found."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Get selected connection and go to the connect step.
+            self._connection = next(
+                (
+                    conn
+                    for conn in self._discovered_connections
+                    if conn.name == user_input["select_receiver"]
+                ),
+                None,
+            )
+            return await self.async_step_connect()
+
+        select_scheme = vol.Schema(
+            {
+                vol.Required("select_receiver"): vol.In(
+                    [conn.name for conn in self._discovered_connections]
+                )
+            }
+        )
+
+        return self.async_show_form(
+            step_id="select", data_schema=select_scheme, errors=errors
+        )
+
+    async def async_step_connect(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Connect to the network receiver."""
+        conn: Connection = self._connection
+        await self.async_set_unique_id(conn.identifier)
+        self._abort_if_unique_id_configured()
+
+        if not await _validate_connection(conn):
+            return self.async_abort(reason="cannot_connect")
+
+        return self.async_create_entry(
+            title=conn.name,
+            data={
+                CONF_IDENTIFIER: conn.identifier,
+                CONF_HOST: conn.host,
+                CONF_NAME: conn.name,
+            },
+        )
+
+    def get_unique_connections(self, connections: list[Connection]) -> list[Connection]:
+        """Filter out connections that are already set up."""
+        return [
+            conn
+            for conn in connections
+            if conn.identifier
+            not in [
+                entity.unique_id
+                for entity in self._async_current_entries(include_ignore=True)
+            ]
+        ]
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
From what I can see having these at the level forces the user to use the same input sources across all of their receivers.

I don't think it allows the following to happen: 

Receiver 1 - I want to import Game, TV and rename Game to "PlayStation"
Receiver 2 - I want to import Game, CD, TV, Aux and rename Game to "XBox"